torch/distributed/fsdp/_fully_shard/_fsdp_collectives.py

```diff
@@ -95,6 +96,7 @@ def all_gather_copy_in_cuda(
 
 @torch.library.impl(lib, "split_with_sizes_copy", "Meta")
 @torch.library.impl(lib, "split_with_sizes_copy", "CUDA")
+@torch.library.impl(lib, "split_with_sizes_copy", "XPU")
```

Comment:
By the way, if you do not implement a fast kernel for `split_with_sizes_copy.out` and `_chunk_cat`, then FSDP2 will not be as performant on XPU. These kernels are required on CUDA for good performance parity with FSDP1.

Reply:
@awgu Thanks for your comment. We also notice such CUDA fast pass in https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L14757 and https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L5720. 

You are right that we have a plan to have same support like CUDA for a better performance. In this PR, we propose some changes in Python frontend to enable FSDP2 functionality on XPU device (otherwise, this feature cannot work on XPU). After that, performant kernels will be implemented.